src/main/java/net/sf/jabref/logic/integrity/IntegrityCheck.java

```diff
@@ -43,36 +48,83 @@
             return result;
         }
 
-        entry.getFieldOptional("author").ifPresent(data -> AUTHOR_NAME_CHECKER.check(data, "author", entry, result));
+        entry.getFieldOptional("author").ifPresent(data -> AUTHOR_NAME_CHECKER.check(data, "author", entry, result, bibDatabaseContext));
 
-        entry.getFieldOptional("editor").ifPresent(data -> AUTHOR_NAME_CHECKER.check(data, "editor", entry, result));
+        entry.getFieldOptional("editor").ifPresent(data -> AUTHOR_NAME_CHECKER.check(data, "editor", entry, result, bibDatabaseContext));
 
         entry.getFieldOptional("title").ifPresent(data -> {
-            if(!isBiblatexMode) {
-                TITLE_CHECKER.check(data, "title", entry, result);
+            if (!bibDatabaseContext.isBiblatexMode()) {
+                TITLE_CHECKER.check(data, "title", entry, result, bibDatabaseContext);
             }
-            BRACKET_CHECKER.check(data, "title", entry, result);
+            BRACKET_CHECKER.check(data, "title", entry, result, bibDatabaseContext);
         });
 
-        entry.getFieldOptional("year").ifPresent(data -> YEAR_CHECKER.check(data, "year", entry, result));
+        entry.getFieldOptional("year").ifPresent(data -> YEAR_CHECKER.check(data, "year", entry, result, bibDatabaseContext));
+
+        entry.getFieldOptional("pages").ifPresent(data -> PAGES_CHECKER.check(data.toString(), "pages", entry, result, bibDatabaseContext));
+
+        entry.getFieldOptional("url").ifPresent(data -> URL_CHECKER.check(data, "url", entry, result, bibDatabaseContext));
+
+        entry.getFieldOptional("file").ifPresent(data -> FILE_CHECKER.check(data, "file", entry, result, bibDatabaseContext));
 
-        entry.getFieldOptional("pages").ifPresent(data -> PAGES_CHECKER.check(data.toString(), "pages", entry, result));
+        entry.getFieldOptional("pages").ifPresent(data -> TYPE_CHECKER.check(data, "pages", entry, result, bibDatabaseContext));
 
-        entry.getFieldOptional("url").ifPresent(data -> URL_CHECKER.check(data, "url", entry, result));
+        entry.getFieldOptional("journal").ifPresent(data -> ABBREVIATION_CHECKER.check(data, "journal", entry, result, bibDatabaseContext));
+        entry.getFieldOptional("booktitle").ifPresent(data -> ABBREVIATION_CHECKER.check(data, "booktitle", entry, result, bibDatabaseContext));
 
         return result;
     }
 
     public interface Checker {
+        void check(String value, String fieldName, BibEntry entry, List<IntegrityMessage> collector, BibDatabaseContext bibDatabaseContext);
```

Comment:
The problem is that the ProceedingsChecker is the exception, not the rule. With your change, the ProceedingsChecker would be more readable, but I would loose the statelessness of the checkers. I am not sure, if this improves the situation. What is more, the check whether the field is there as to be implemented in each checker over and over again, creating noise in the check logic. 

Passing the list as a collector I also like because otherwise one has to create a lot of lists and combine them. The collector pattern is used by most test frameworks as well, e.g., JUnit does this internally.